### PR TITLE
Add a way to explicity choose aggregators in QB

### LINKED ISSUE #4620: Add support for explicitly choosing an aggregator in Query Builder
The feature request "Add support for explicitly choosing a formatter in Query Builder" (https://github.com/specify/specify7/issues/1959) did not cover choosing an aggregator.

@maxpatiiuk said in [this comment](https://github.com/specify/specify7/issues/1959#issuecomment-1579769627) that at one point it was
>Fixed both for formatters and aggregators

[This code](https://github.com/specify/specify7/commit/231ac56ca81e24debfb3d54b15426bdde39edc45#diff-beffe9e3545186073f701dcc4e18ee942db4e7f7a0439fec2c3fdb8d5f281e26R376-R396) conditionally renders a `QueryFieldFormatter` component based on the `fieldMeta` properties. If `fieldType` is either 'formatter' or 'aggregator' and `tableName` is a string, it displays the `QueryFieldFormatter` component with specific props like `formatter`, `tableName`, `type`, and `onChange`.

Theoretically it already _should_ support this for aggregators, but either due to a bug in the existing code or an oversight, this is not functional at the moment.

This indicates to me that the intention was to have this existing feature request cover aggregators as well (as the logic should be nearly identical).

**Expected Behavior**
* When adding an `(aggregated)` line to a query, the user should be able to explicitly define which aggregation to use
* The gear icon should only appear to the right of the `(aggregated)` line if there are two or more aggregators configured for that table
* This should behave **identically** to how the user defines an explicit formatter (added back in https://github.com/specify/specify7/pull/4615)

This is arguably a bug in the existing implementation, but because it is not functional and outside the scope of the previous issue, I have opened this one. I do see this as one and the same as #1959 at its core.

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4620

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

- Navigate to the Query Builder (QB).
- Select a table that you know has one or several aggregators (To check, go to app resources, navigate to Record Formatter, choose the table, select aggregator tab, and verify the list of aggregators names available).
- Choose "aggregated."

- [ ] Verify that in the query line, after selecting the "aggregated" field box, a drop-down appears with a list of aggregators choices (corresponding to the list in app resources).

 Verify that changing the aggregator changes the way the field is displayed in the results table.

### CODE REVIEW

**Comment by acwhite211** [comment]
The problem seems to be that the back-end is ignoring the formatter name in the request.  It is just getting the aggragator and then getting the formatter that the aggregator is pointing to.  I changed the code so that it takes into account the 'formatter name' that is passed in the query builder request.  I just need to fix some of the unit tests now.

File: specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx [hunk 1]
```diff
@@ -167,7 +166,7 @@ export function QueryLine({
         canOpenMap = fieldName === 'latitude1' || fieldName === 'longitude1';
       } else if (isMapped)
         fieldType =
-          dataModelField?.isRelationship && relationshipIsToMany(dataModelField)
+          isFormatted && mappingPath.at(-1) === '#1'
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx:169]: could you please apply the fixes I mentioned in https://github.com/specify/specify7/pull/4615#pullrequestreview-1932351672 in here?

**Review by maxpatiiuk — APPROVED** [review]

**Review by emenslin — CHANGES_REQUESTED** [review]
### Testing instructions

- Navigate to the Query Builder (QB).
- Select a table that you know has one or several aggregators (To check, go to app resources, navigate to Record Formatter, choose the table, select aggressors tab, and verify the list of aggregators names available).
- Choose "aggregated."

- [x] Verify that in the query line, after selecting the "aggregated" field box, a drop-down appears with a list of aggregators choices (corresponding to the list in app resources).

- [ ] Verify that changing the aggregator changes the way the field is displayed in the results table.

Changing the aggregator does not change the way the field is displayed and seems to only show the default aggregation. 

https://github.com/specify/specify7/assets/150940840/3a54fd5f-36dd-4135-a86b-b139a304493f

**Review by emenslin — APPROVED** [review]
### Testing instructions

- Navigate to the Query Builder (QB).
- Select a table that you know has one or several aggregators (To check, go to app resources, navigate to Record Formatter, choose the table, select aggregator tab, and verify the list of aggregators names available).
- Choose "aggregated."

- [x] Verify that in the query line, after selecting the "aggregated" field box, a drop-down appears with a list of aggregators choices (corresponding to the list in app resources).
- [x] Verify that changing the aggregator changes the way the field is displayed in the results table.

Looks good, changing the aggregator changes the results now!

https://github.com/specify/specify7/assets/150940840/6c9ba09a-391c-4283-8f09-b19e49809895

**Review by alesan99 — APPROVED** [review]
- [x] Verify that in the query line, after selecting the "aggregated" field box, a drop-down appears with a list of aggregators choices (corresponding to the list in app resources).

Added new aggregators and it looks good, & changes are reflected in Query results

chrome_8GE0d2wOxq

**Pull request merged**